plugins/modules/aci_access_switch_policy_group.py

```diff
@@ -0,0 +1,472 @@
+        poe_node_policy=dict(type="str"),
+        fibre_channel_san_policy=dict(type="str"),
+        monitoring_policy=dict(type="str"),
+        netflow_node_policy=dict(type="str"),
+        copp_leaf_policy=dict(type="str"),
+        forward_scale_profile_policy=dict(type="str"),
+        fast_link_failover_policy=dict(type="str"),
+        node_802_1x_authentication_policy=dict(type="str"),
+        copp_pre_filter_policy=dict(type="str"),
+        equipment_flash_policy=dict(type="str"),
+        cdp_policy=dict(type="str"),
+        lldp_policy=dict(type="str"),
+        sync_e_node_policy=dict(type="str"),
+        ptp_node_policy=dict(type="str"),
+        usb_configuration_policy=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name"]],
+        ],
+    )
+    name = module.params.get("name")
+    description = module.params.get("description")
+    spanning_tree_policy = module.params.get("spanning_tree_policy")
+    bfd_ipv4_policy = module.params.get("bfd_ipv4_policy")
+    bfd_ipv6_policy = module.params.get("bfd_ipv6_policy")
+    bfd_multihop_ipv4_policy = module.params.get("bfd_multihop_ipv4_policy")
+    bfd_multihop_ipv6_policy = module.params.get("bfd_multihop_ipv6_policy")
+    fibre_channel_node_policy = module.params.get("fibre_channel_node_policy")
+    poe_node_policy = module.params.get("poe_node_policy")
+    fibre_channel_san_policy = module.params.get("fibre_channel_san_policy")
+    monitoring_policy = module.params.get("monitoring_policy")
+    netflow_node_policy = module.params.get("netflow_node_policy")
+    copp_leaf_policy = module.params.get("copp_leaf_policy")
+    forward_scale_profile_policy = module.params.get("forward_scale_profile_policy")
+    fast_link_failover_policy = module.params.get("fast_link_failover_policy")
+    node_802_1x_authentication_policy = module.params.get("node_802_1x_authentication_policy")
+    copp_pre_filter_policy = module.params.get("copp_pre_filter_policy")
+    equipment_flash_policy = module.params.get("equipment_flash_policy")
+    cdp_policy = module.params.get("cdp_policy")
+    lldp_policy = module.params.get("lldp_policy")
+    sync_e_node_policy = module.params.get("sync_e_node_policy")
+    ptp_node_policy = module.params.get("ptp_node_policy")
+    usb_configuration_policy = module.params.get("usb_configuration_policy")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="infraAccNodePGrp",
+            aci_rn="infra/funcprof/accnodepgrp-{0}".format(name),
+            module_object=name,
+            target_filter={"name": name},
+        ),
+        rsp_subtree="children",
```

Comment:
Why are we deviating from using child_classes?

Reply:
Because the child_classes are different based on aci version. When executing the query with all child_classes it would fail on the get request because the child class does not exists.

Originally Tim created a function to query the version and based on version execute the get request with the correct child_classes. This would introduce additional api call and version check function. To avoid this we can query the with rso_subtree set to children to retrieve all child classes.

Reply:
I was thinking of having a child_classes similar to child_configs and appending a child class to it only if the corresponding attribute was specified.
Example:  
```
child_configs = []
child_classes = []
if forward_scale_profile_policy is not None:
            child_classes.append("infraRsTopoctrlFwdScaleProfPol")
            child_configs.append({"infraRsTopoctrlFwdScaleProfPol": {"attributes": {"tnTopoctrlFwdScaleProfilePolName": forward_scale_profile_policy}}})
if usb_configuration_policy is not None:
            child_classes.append("infraRsLeafTopoctrlUsbConfigProfilePol")
            child_configs.append({"infraRsLeafTopoctrlUsbConfigProfilePol": {"attributes": {"tnTopoctrlUsbConfigProfilePolName": usb_configuration_policy}}})
```

However, if the change in aci.py yields the same results and has no side-effects your solution is better. 
  ```
  elif rsp_subtree == "children":
            self.update_qs({"rsp-subtree": rsp_subtree})
```

Reply:
But this would restrict the information displayed for querying all the children. Resolving this.